examples/multiple_errors.rs

```diff
@@ -0,0 +1,62 @@
+//! It's not uncommon for programs like batched task runners or parsers to want to return an error
+//! with multiple sources. The current version of the error trait does not support this usecase
+//! very well, though there is work being done to improve this.
+//!
+//! For now however one way to work around this is to compose errors outside the error trait.
+//! `color-eyre` supports such composition in its error reports via its custom sections feature.
+//!
+//! This example is intended to demonstrate two approaches to error aggregation with color-eyre
+
+use color_eyre::{Help, Report};
+use eyre::eyre;
+use thiserror::Error;
+
+fn main() -> Result<(), Report> {
+    let errors = get_errors();
+    join_errors(errors)
+}
+
+fn join_errors(results: Vec<Result<(), SourceError>>) -> Result<(), Report> {
+    if results.iter().all(|r| r.is_ok()) {
+        return Ok(());
+    }
+
+    results
+        .into_iter()
+        .filter(Result::is_err)
+        .map(Result::unwrap_err)
+        .fold(Err(eyre!("encountered multiple errors")), |report, e| {
+            report.error(e)
+        })
+}
+
+/// Helper function to generate errors
+fn get_errors() -> Vec<Result<(), SourceError>> {
+    vec![
+        Err(SourceError {
+            source: StrError("The task you ran encountered an error"),
+            msg: "The task could not be complete",
+        }),
+        Err(SourceError {
+            source: StrError("The machine you're connecting too is actively on fire"),
+            msg: "the machine is unreachable",
```

Comment:
```suggestion
            source: StrError("The machine you're connecting to is actively on fire"),
            msg: "The machine is unreachable",
```

Are you familiar with https://en.wikipedia.org/wiki/Lp0_on_fire? :D

Reply:
I have not but I'm excited to read that now